backend/src/shortener/shortener.service.spec.ts

```diff
@@ -93,24 +95,93 @@ describe('ShortenerService', () => {
     const shortUrl = 'best_url_shortener';
 
     await service.addUrl(originalUrl, shortUrl);
-    expect(async () => {
+    await expect(async () => {
       await service.addUrl(originalUrl, shortUrl);
-    }).rejects.toThrowError('Short URL already taken');
+    }).rejects.toThrowError('ShortenedURL already taken');
   });
 
   describe('isUrlAlreadyShortend', () => {
     it('should return true because the url is already shortened', async () => {
       const shortenUrl = `${config.getConfig().front.domain}/test123`;
 
-      const result = service.isUrlAlreadyShortend(shortenUrl);
+      const result = service.isUrlAlreadyShortened(shortenUrl);
       expect(result).toBeTruthy();
     });
 
     it('should return false because the url is not shortened', async () => {
       const url = 'https://github.com/origranot/reduced.to';
 
-      const result = service.isUrlAlreadyShortend(url);
+      const result = service.isUrlAlreadyShortened(url);
       expect(result).toBe(false);
     });
   });
+
+  describe('getUrlFromDb', () => {
+    it('should return the premium url', async () => {
+      prisma.url.findFirst = jest.fn().mockReturnValueOnce({
+        originalUrl: 'original_url',
+      });
+      const result = await service.getUrlFromDb('good_url');
+      expect(result).toBe('original_url');
+    });
+
+    it('should return null if the premium url is not found', async () => {
+      prisma.url.findMany = jest.fn().mockReturnValueOnce(undefined);
+      const result = await service.getUrlFromDb('expired_url');
+      expect(result).toBe(null);
+    });
+  });
+
+  describe('getShortUrl', () => {
+    it('should return a shortern url', async () => {
+      jest.spyOn(service, 'generateShortenedUrl').mockReturnValue('best');
+      jest.spyOn(service, 'isShortenedUrlAvailable').mockResolvedValue(true);
+      jest.spyOn(service, 'addUrl').mockResolvedValue(undefined);
+      jest.spyOn(service, 'isUrlAlreadyShortened').mockReturnValue(false);
+
+      const body: ShortenerDto = { originalUrl: 'https://github.com/origranot/reduced.to' };
+      const short = await service.createShortUrl(body);
+      expect(short).toStrictEqual({ newUrl: 'best' });
+    });
+
+    it('should throw an error of invalid url', () => {
```

Comment:
Wrap this with describe of createShortUrl

Reply:
done